staging/src/k8s.io/client-go/plugin/pkg/client/auth/exec/exec.go

```diff
@@ -191,7 +191,15 @@ func (a *Authenticator) UpdateTransportConfig(c *transport.Config) error {
 		dial = (&net.Dialer{Timeout: 30 * time.Second, KeepAlive: 30 * time.Second}).DialContext
 	}
 	d := connrotation.NewDialer(dial)
+
+	a.mu.Lock()
+	defer a.mu.Unlock()
+	// Before replace with new connrotation dialer, close all existing connections.
+	if a.onRotate != nil {
+		a.onRotate()
```

Comment:
The linked issue indicates that a single Authenticator instance is being shared between all clients constructed from the same config, which I guess is expected because of the authenticator config cache:

https://github.com/kubernetes/kubernetes/blob/81a8b9804a3bf310cb74446bd8a8e08e0317de22/staging/src/k8s.io/client-go/plugin/pkg/client/auth/exec/exec.go#L113-L124

Calling onRotate here means we will close all connections opened by this authenticator, which means constructing a new client from a config will disrupt all previous clients constructed from that config, which we don't want.

Overriding a.onRotate here means that we lose the ability to close connections for previously constructed clients, which we also don't want.

The most straightforward change to make here (without restructuring the 1:many relationship between Authenticator and Clients instances) is to:
1. make onRotate a slice
2. change this override to an append (and do not call any existing onRotate here)
3. change the call to onRotate in refreshCredsLocked to iterate and call all onRotate methods

Note that if the calling code repeatedly constructs clients (instead of constructing them once and reusing them), that has the potential to grow the onRotate slice without bound.

Reply:
SGTM.
PR updated.
> Note that if the calling code repeatedly constructs clients (instead of constructing them once and reusing them), that has the potential to grow the onRotate slice without bound.

How about these two method:
1. set the maximum of onRotateList
2. print warning log if the length of onRotateList exceeds the maximum number

This PR uses the second one.